### DISCUSSION TITLE
请问我的代码既有python2，又有python3，应该买版本7，能不能支持设备离线加密？（My code has both python2 and python3. I should buy version 7. Can I support offline encryption?）

### DISCUSSION THREAD

**Discussion by PhilipWang34** [dashingsoft/pyarmor#1778] [Q&A]
请问我的代码既有python2，又有python3，应该买版本7，能不能支持设备离线加密？（My code has both python2 and python3. I should buy version 7. Can I support offline encryption?）

**Comment by jondy** [comment]
Pyarmor 7 支持 Python2.7~Python3.10, 也支持离线加密，但是不支持 Python3.11+
